src/attributes/codegen.md

```diff
@@ -374,4 +375,9 @@ fn foo_arm_code() {}
 fn bar_thumb_code() {}
 ```
 
-[_MetaListPath_]: ../attributes.md#meta-item-attribute-syntax
+If your function has neither the `instruction_set` attribute nor inline assembly, then the code you write within that function should not presume any particular instruction set.
+This ends up creating a limitation to how often code is inlined:
+
+* If a function has an `instruction_set` attribute it won't inline into a function of another instruction set.
+* If a function does not have an `instruction_set` attribute but *does* contain inline assembly, then the inline assembly is assumed to require the default instruction set of the build target, and so inlining between different instruction sets won't happen.
+* Otherwise, inlining happens normally.
```

Comment:
This seems like it's specific to ARM and Thumb; this may not necessarily *universally* be true for all future uses of `instruction_set`.